cpp/src/arrow/compute/key_map_internal.cc

```diff
@@ -297,8 +289,9 @@ uint64_t SwissTable::num_groups_for_resize() const {
   }
 }
 
-uint64_t SwissTable::wrap_global_slot_id(uint64_t global_slot_id) const {
-  uint64_t global_slot_id_mask = (1 << (log_blocks_ + 3)) - 1;
+uint32_t SwissTable::wrap_global_slot_id(uint32_t global_slot_id) const {
+  uint32_t global_slot_id_mask =
+      static_cast<uint32_t>((1ULL << (log_blocks_ + kLogSlotsPerBlock)) - 1ULL);
```

Comment:
Interestingly we're still using a 64-bit unsigned shift here (see previous comment).

Reply:
Ahah, ok. Thanks for the explanation!